### ISSUE TITLE
Discrepancy between ice in barrel and in resource inventory and ice aboard vehicle

### ISSUE THREAD

**Issue by mokun** [mars-sim/mars-sim#970]
**Is your feature request related to a problem? Please describe.**
- How do we present the actual amount of ice collected in an Ice Prospecting/Collection mission ?
- From Vehicle Unit Window's Inventory tab, the equipment table shows a total of 11 barrels, each having 57.8 kg of ice, totaling 635.8 kg (11 * 57.8).
- However, in the same Inventory tab, on Resource table, it shows 542.45 kg.
- Yet in Mission Tool's Mission Specific Panel, it shows 206.4 kg ice aboard vehicle.
- Which is true ?

**Describe the solution you desire to see**
- It's possible to show the breakdown of where the ice is stored at, namely, (1) inside barrel and/or (2) in vehicle's specialized storage, using Mission Tool's Main tab and Mission Specific panel.

**Screenshots**
![image](https://github.com/mars-sim/mars-sim/assets/1168584/337b7fcf-61a0-4184-8724-28dacfd5da1a)

**Specifications  (please complete the following information):**
 - pre 3.6.0 build 8394
 
**Additional context**
- See `CollectResourcesMissionCustomInfoPanel`, `CollectResourcesMission`

**Comment by mokun** [comment]
Now that the mission is complete, I do see that the Resource Collected has increased to 542.5 kg in Mission Tool.

![image](https://github.com/mars-sim/mars-sim/assets/1168584/94fe8c05-7e99-4e6a-8198-43a22743ad08)

However, earlier, when I tried to get an estimate of the ice in those barrels, they did add up to over 635+ kg. That's what I still can't explain.

**Comment by mokun** [comment]
@bevans2000 

Can we now dial down the onset of ice/regolith collection missions ? 

So many of them are generated if I run the default scenario of 12 settlements.

**Comment by bevans2000** [comment]
Yes, we need to adjust the number of a active missions per Settlement

**Comment by bevans2000** [comment]
The mission value is only updated at certain points and not all the to e.

**Comment by bevans2000** [comment]
I wonder if we should do something radical about the Mission Probability. Currently we just return a single value and have no idea what the various part values are. But should be create a new ```Rating``` class that holds the final score but also the contributing parts as name value pairs. Then we could dump the score to a file or in the console for detailed analysis/tuning. 
The same could be applied to lots of other places:

- Task probability
- Exploration Site scoring
- Mining Site scoring

Then we could deliver a common class to dump this to a diagnostic file. What do you think? Need a dedicated issue but might help us explain more why certain entities are chosen. Could try and do it as part of 3.6.0 but probably do it for the next release as I'm focussing on closing down the open issues for 3.6.0

**Comment by mokun** [comment]
> we should do something radical about the Mission Probability. Currently we just return a single value and have no idea what the various part values are.

Do you mean saving the mission probability of each mission that is under review ? 

Is it needed ?

**Comment by bevans2000** [comment]
No I was just think of the compenent parts during the mission selection. However we could hold the rating of approved Missions. My thought was it is difficult currently to understand why missions are selected. So for example is the collect missions the only choice or does it always have a high score? This would influence what the fix is

**Comment by mokun** [comment]
> it is difficult currently to understand why missions are selected


Sure.  That's why in another thread, I mentioned we will need to find a way to present all the modifiers to players. 

Usually Civilization and other strategy games show modifiers that abstractly represent and inform players how vertical adjustment would impact activities or production.

Changing a policy or taking an action for or against a cause will change these modifierd right in front of their eyes.

This will greatly improve the interactivity of mars-sim.

**Comment by bevans2000** [comment]
Which thread us that then? I agree showing the modifiers will improve the experience. Assign the other thread to me and I can work on it

**Comment by mokun** [comment]
> Assign the other thread to me and I can work on it

This one at https://github.com/mars-sim/mars-sim/issues/611.

**Comment by mokun** [comment]
For instance, in a PC game called Victoria 3, it shows how the Total Population Attraction ends up being 20 points by listing out a sum of Attraction modifiers ( + 50 + 10 + 10 + 10 - 40 - 20) : 

![image](https://github.com/mars-sim/mars-sim/assets/1168584/5a022f82-8403-41cb-9cd7-272b4012e53a)

**Comment by bevans2000** [comment]
Rating idea captured as #978
